patches/api/0376-Add-EquipmentSlot-convenience-methods.patch

```diff
@@ -24,6 +24,21 @@ index 5642d8af60b6649497aba9b0f6ab7bba7702b9ee..adf0a94f98eb66b3957d4009d83ad5f7
 +    }
 +
 +    /**
++     * Gets the opposite hand
++     *
++     * @return the opposite hand
++     * @throws IllegalStateException if this equipment slot is not a hand
++     * @see #isHand()
++     */
++    public @NotNull EquipmentSlot getOppositeHand() {
++        return switch (this) {
++            case HAND -> OFF_HAND;
++            case OFF_HAND -> HAND;
++            default -> throw new IllegalStateException("Can't get an opposite hand for not a hand");
```

Comment:
Maybe something like?
```suggestion
+            default -> throw new IllegalStateException("Unable to determine opposite hand for hand: " + name());
```

Reply:
It's not gonna be a hand, so the end of the sentence doesn't work. Though also: that should be IllegalArgumentException or something, the state one implies the application is in the wrong state to accept a query

Reply:
```suggestion
+            default -> throw new IllegalArgumentException("Unable to determine opposite hand for equipment slot: " + name());
```
Something like this?

Reply:
On a second thought is it necessary to throw an exception? Can't we just return null? Exception seems a little bit overkill, no?

Reply:
returning null is generally crass, you're performing an operation which is enacting on an illegal argument to itself

Reply:
Added name() to the output and changed to IllegalArgumentException.
Initially went with IllegalStateException due to exceptions' javadocs interpretation, they are somewhat vague.

Reply:
"For not a hand" still sounds weird. Why don't we replace it with "for equipment slot"?